### PR TITLE
fix: wovalle#183 batch write within subcollection

### PR DESCRIPTION
**Pull request by Fame513**
Hi, @skalashnyk asked me to help him with the issue #183. I fixed that problem, and added the case to the tests.

### CODE REVIEW

File: test/functional/3-subcollections.int-spec.ts [hunk 1]
```diff
@@ -14,7 +8,7 @@ describe('Integration test: SubCollections', () => {
   @Collection(getUniqueColName('band-with-subcollections'))
   class FullBand extends BandEntity {
     @SubCollection(Album)
-    albums: ISubCollection<Album>;
+    albums: BaseFirestoreRepository<Album>;
```
Comment (wovalle) [inline test/functional/3-subcollections.int-spec.ts:11]: Why did you change this type? Ideally it should be a `IRepository`? I remember that I had some problems with #172  but can't recall exactly what it was

**Review by wovalle — APPROVED** [review]
Great job! I just have a question before releasing.

File: test/functional/3-subcollections.int-spec.ts (hunk 1 above)
Comment (skalashnyk) [reply test/functional/3-subcollections.int-spec.ts:11]: 1) `ISubCollection` interface (that is alias of `IRepository`) does not expose `createBatch()` method - https://github.com/wovalle/fireorm/blob/1ebfde9aa983a7f0068251eb11a64a583cfb8d0b/src/types.ts#L80
2) As we can see, subcollection property becomes instance of `BaseFirestoreRepository` while deserialization:
https://github.com/wovalle/fireorm/blob/1ebfde9aa983a7f0068251eb11a64a583cfb8d0b/src/AbstractFirestoreRepository.ts#L88 and https://github.com/wovalle/fireorm/blob/1ebfde9aa983a7f0068251eb11a64a583cfb8d0b/src/helpers.ts#L13

Comment (wovalle) [reply test/functional/3-subcollections.int-spec.ts:11]: Thanks! @skalashnyk, makes sense. I'll have to check the typing ISubCollection type declaration, for now is ok

**Comment by wovalle** [comment]
@all-contributors please add @skalashnyk and @Fame513 for code

**Comment by wovalle** [comment]
@all-contributors please add @Fame513 for code
